shared/src/main/scala/scodec/Attempt.scala

```diff
@@ -75,6 +80,16 @@ object Attempt {
   /** Creates an unsuccessful attempt. */
   def failure[A](err: Err): Attempt[A] = Failure(err)
 
+  /** Creates a successful attempt if the condition succeeds otherwise create a unsuccessful attempt */
+  def guard(condition: => Boolean, err: String) =
+    if(condition) successful(()) else failure(Err(err))
+
+  /** Creates a attempt from a try */
```

Comment:
What is period? Is it `@since` or `@version` as scaladoc ? :-) If so, what version?